Yank/pipeline.py

```diff
@@ -1417,6 +1417,182 @@ def _setup_system(self, system_file_path, pack, alchemical_charge,
             logger.warning('TLeap: ' + warning)
 
 
+# ==============================================================================
+# FUNCTIONS FOR ALCHEMICAL PATH OPTIMIZATION
+# ==============================================================================
+
+def restrain_atoms(thermodynamic_state, sampler_state, mdtraj_topology,
```

Comment:
I'll move it to `openmmtools.forcefactories`. We'll need to make an `openmmtools` release before merging this then and update the minimum version in the conda recipe.